catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/cache/solr/impl/SolrCache.java

```diff
@@ -292,6 +294,16 @@ public void removeById(String[] ids) throws IOException, SolrServerException {
     return getMetacardsFromResponse(response);
   }
 
+  public void setAnyTextAttributes(List<String> anyTextAttributes) {
+    this.anyTextAttributes.clear();
+    if (CollectionUtils.isNotEmpty(anyTextAttributes)) {
+      this.anyTextAttributes.addAll(anyTextAttributes);
+    }
+    if (metacardClient instanceof CacheSolrMetacardClient) {
+      ((CacheSolrMetacardClient) metacardClient).setAnyTextAttributes(anyTextAttributes);
```

Comment:
🤔 This seems like a smell to me if we have to check the implementation of the client ...

Reply:
Shouldn't the `instanceof` check be done before the attributes are cleared?